server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/ToSenderDomainRepository.java

```diff
@@ -0,0 +1,89 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.transport.mailets;
+
+import java.util.Optional;
+
+import javax.inject.Inject;
+import javax.mail.MessagingException;
+
+import org.apache.james.mailrepository.api.MailRepositoryStore;
+import org.apache.mailet.Mail;
+import org.apache.mailet.base.GenericMailet;
+
+/**
+ * Stores incoming Mail in a repository defined by the sender's domain.<br>
+ *
+ * Supported configuration parameters:
+ *
+ *  - "urlPrefix" compulsory: defines the prefix for the per sender's domain repository.
+ *  If equals to 'cassandra://var/mail/sendersRepositories/' a mail sent by 'user@domain.tld'
+ *  will be stored in 'cassandra://var/mail/sendersRepositories/domain.tld'.
```

Comment:
do we need to have an option to create the repository if not already existing?

Reply:
For the record @rouazana  disabling repository creation is pointless if the admin can't create the mail repositories he wants. Hence I asked Duc to add https://github.com/linagora/james-project/pull/1457/commits/f0ee57225518c64e4a3f6e6e57b1efb76d51c5c5